packages/logger/tests/e2e/basicFeatures.middy.test.ts

```diff
@@ -0,0 +1,165 @@
+    }, TEST_CASE_TIMEOUT);
+  });
+
+  describe('Context data', () => {
+    it('should log context information of the function', async () => {
+      const logMessages = invocationLogs[0].getFunctionLogs();
+
+      for (const message of logMessages) {
+        expect(message).toContain('function_arn');
+        expect(message).toContain('function_memory_size');
+        expect(message).toContain('function_name');
+        expect(message).toContain('function_request_id');
+        expect(message).toContain('timestamp');
+      }
+    }, TEST_CASE_TIMEOUT);
+
+    it('should include cold metrics only on the first invocation', async () => {
+      const coldStartLogMessages = invocationLogs[0].getFunctionLogs(LEVEL.INFO);
+      for (const message of coldStartLogMessages) {
+        expect(message).toContain(`"cold_start":true`);
+      }
+      
+      // TODO: There is a bug that the 'cold_start' field leaks to all invocation. (#518) 
+      // Link to tracked issue: https://github.com/awslabs/aws-lambda-powertools-typescript/issues/518
+      // const normalLogMessages = invocationLogs[1].getFunctionLogs(LEVEL.INFO);
+      // for(const message of normalLogMessages) {
+      //   expect(message).not.toContain(`"cold_start":true`);
+      // }
+    }, TEST_CASE_TIMEOUT);
+  });
+
+  describe('Context data', () => {
+    it('should log context information in every log', async () => {
+      const logMessages = invocationLogs[0].getFunctionLogs();
+
+      for (const message of logMessages) {
+        expect(message).toContain('function_arn');
+        expect(message).toContain('function_memory_size');
+        expect(message).toContain('function_name');
+        expect(message).toContain('function_request_id');
+        expect(message).toContain('timestamp');
+      }
+    }, TEST_CASE_TIMEOUT);
+  });
+
+  describe('Persistent additional log keys and values', () => {
+    it('should contain persistent value in every log', async () => {
+      const logMessages = invocationLogs[0].getFunctionLogs();
+
+      for (const message of logMessages) {
+        expect(message).toContain(`"${PERSISTENT_KEY}":"${PERSISTENT_VALUE}"`);
+      }
+    }, TEST_CASE_TIMEOUT);
+  });
+
+  describe('One-time additional log keys and values', () => {
+    it('should log additional keys and value only once', async () => {
+      const logMessages = invocationLogs[0].getFunctionLogs()
+        .filter(message => message.includes(`"${SINGLE_LOG_ITEM_KEY}":"${SINGLE_LOG_ITEM_VALUE}"`));
+      
+      expect(logMessages).toHaveLength(1);
+    }, TEST_CASE_TIMEOUT);
+  });
+
+  describe('Logging an error object', () => {
+    it('should log additional keys and value only once', async () => {
+      const logMessages = invocationLogs[0].getFunctionLogs(LEVEL.ERROR)
+        .filter(message => message.includes(ERROR_MSG));
+      
+      expect(logMessages).toHaveLength(1);
+
+      // Each log contains:
+      // We will extract the last part
```

Comment:
Could you elaborate more on these comments? From the way it reads it seems a line is missing